pkg/cluster/validate_test.go

```diff
@@ -75,3 +76,52 @@ func TestValidateConfigDifferentNamespace(t *testing.T) {
 		MatchError(ContainSubstring("VSphereDatacenterConfig and Cluster objects must have the same namespace specified")),
 	)
 }
+
+func TestValidateCloudstackConfigDifferentNamespace(t *testing.T) {
+	g := NewWithT(t)
+	tests := []struct {
+		name      string
+		clusterNS string
+		dcNS      string
+		mcNS      string
+		wantErr   string
+	}{
+		{
+			name:      "cluster, datacenter config, machine config in the same namespace",
+			clusterNS: constants.EksaSystemNamespace,
+			dcNS:      constants.EksaSystemNamespace,
+			mcNS:      constants.EksaSystemNamespace,
+			wantErr:   "",
+		},
+		{
+			name:      "datacenter config not at the same namespace",
+			clusterNS: constants.EksaSystemNamespace,
+			dcNS:      "dc-ns",
+			mcNS:      constants.EksaSystemNamespace,
+			wantErr:   "CloudStackDatacenterConfig and Cluster objects must have the same namespace specified",
+		},
+		{
+			name:      "machine config not at the same namespace",
+			clusterNS: constants.EksaSystemNamespace,
+			dcNS:      constants.EksaSystemNamespace,
+			mcNS:      "mc-ns",
+			wantErr:   "CloudStackMachineConfig and Cluster objects must have the same namespace specified",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			config := clusterConfigFromFile(t, "testdata/cluster_1_20_cloudstack.yaml")
```

Comment:
Since we are only testing for the namespace, can we create the config through code like the above function rather than a file? In general we want to remove the number of references to files that we have